src/v/archival/ntp_archiver_service.cc

```diff
@@ -995,8 +1001,13 @@ ntp_archiver::schedule_single_upload(const upload_context& upload_ctx) {
     // The upload is successful only if both segment and tx_range are uploaded.
     std::vector<ss::future<cloud_storage::upload_result>> all_uploads;
     all_uploads.emplace_back(upload_segment(upload));
+    size_t tx_size = 0;
     if (upload_ctx.upload_kind == segment_upload_kind::non_compacted) {
-        all_uploads.emplace_back(upload_tx(upload));
+        auto tx = co_await get_aborted_transactions(upload);
+        tx_size = tx.size();
+        if (!tx.empty()) {
+            all_uploads.emplace_back(upload_tx(upload, std::move(tx)));
+        }
```

Comment:
I'm not sure I understand why we're gating setting the metadata size hint on whether we're non-compacted. I could be understanding the rationale here wrong, but I thought we don't reupload the tx metadata because the tx manifest already exists in the case of a compaction upload. But if we reset the hint to 0, would that lead to problems?

Reply:
Or perhaps we remove aborted transactions during compactions? Either way, a comment would be nice

Reply:
yep, they are removed and we never downloading tx-manifest if segment is compacted, I added the comment